### PR TITLE
Awesome books- add and delete books from list

### PR DESCRIPTION
**Pull request by lily-coder**
In this project, we have built a basic website that allows users to add/remove books from a list by:
-Creating a collection that keeps a list of books .
-Creating a function to add a new book to the collection, with title and author.
-Creating a function to remove a book from the collection.
-Displaying all books saved in the collection in the top part of the page.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,55 @@
...
+> In this project, we are using Javascript to build a basic website that allows users to add/remove books from a list. 
+
+## Built With
+
+- Html5
+- Javascript
+
+## Live Demo
+
+[Live Demo Link](https://livedemo.com)
```
Comment (o-rashad) [inline README.md]: Kindly fix it to your Actual Project LIVE link if available 🎥

File: README.md [hunk 2]
```diff
@@ -1,2 +1,55 @@
...
+
+### Setup
+To get a local copy up and running follow these simple example steps.
+
+- Clone the repository to your computer using this[link] (https://github.com/menyagah/Awesome-books.git)
```
Comment (o-rashad) [inline README.md]: Something wrong with this link 🤔 ?

File: style.css [hunk 3]
```diff
@@ -0,0 +1,3 @@
+h1 {
+  color: aqua;
+}
```
Comment (o-rashad) [inline style.css]: Is this an unused Style rule ? 🤔
for our small project, it's okay, but at later ones, we gonna need to reduce our unused CSS rules...


```
Why?
Depending on the layout and your visitors' device, loading CSS rules is not needed to render 
above-the-fold content can have a heavy impact on your page performance, both, measured and perceived. 
``` 
[Source](https://gtmetrix.com/reduce-unused-css.html#:~:text=Depending%20on%20the%20layout%20and%20your%20visitors%27%20device%2C%20loading%20CSS%20rules%20not%20needed%20to%20render%20above-the-fold%20content%20can%20have%20a%20heavy%20impact%20on%20your%20page%20performance%2C%20both%2C%20measured%20and%20perceived.)

For now please remove the CSS file if it is unused 😊

File: main.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const title = document.querySelector('.title');
+const author = document.querySelector('.author');
+const button = document.querySelector('.btn');
+
+// REMOVE BOOK FUNCTION
+function deleteBook(el) {
+  if (el.classList.contains('remove')) {
+    el.parentElement.parentElement.remove();
+  }
+}
```
Comment (o-rashad) [inline main.js]: Your Function to delete a book object is working perfectly, but only removing the book from the DOM (disappear from your page).  👍🏼 

However, You need to delete the object from your `books` collection too. as stated here...

e5ecde40-f0ce-4b53-bc05-feadab708ec9

**for demonstration: please observe the book variable in the three situations**

Untitled

File: main.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+// ADD BOOKS TO TABLE LIST
+function addBookList() {
+  const list = document.querySelector('.book-list');
+
+  const row = document.createElement('tr');
+  // RETRIEVE THE BOOK FROM LOCAL STORAGE IN JSON FORMAT
+  window.localStorage.getItem('book');
+  // CONVERT THE JSON FORMAT BACK TO AN OBJECT
+  JSON.parse(window.localStorage.getItem('book'));
```
Comment (o-rashad) [inline main.js]: It would be nice if we get rid of the unused code in our codebase 👌🏼

File: main.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  list.appendChild(row);
+}
+
+// ADD INPUT VALUE
+button.addEventListener('click', () => {
+  books.push({ title: title.value, author: author.value });
+
+  addBookList();
+  window.localStorage.setItem('book', JSON.stringify(books));
```
Comment (o-rashad) [inline main.js]: Very good job implementing the `LocalStorage` API to store the `books` array after adding a new book, 💯  

Sadly when I refresh the page, all the books that I created are not there anymore, which means that is something wrong with retrieving your variable from the `LocalStorage` at the beginning of your App. 

Kindly give it another look. 👀

**Review by o-rashad — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hi @lily-coder , @menyagah 👋😄, I'm Omar and I will be your reviewer for this project 😊

Good job so far!👌😃

There are some issues 🐞 that you still need to work on to go to the next project but you are almost there! 🏁🏁🏁

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.js (hunk 6 above)
Comment (Davidosky007) [reply main.js]: * Done 👍

**Comment by Davidosky007** [comment]
## Hi @lily-coder @menyagah 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**